bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/Strand.java

```diff
@@ -501,7 +501,7 @@ private void getInfoFromYieldedState(StringBuilder strandInfo, String closingBra
             strandInfo.append(RUNNABLE).append(closingBracketWithNewLines);
             return;
         }
-        if (!this.isYielded() || noPickedYieldStatus) {
+        if (!this.isYielded()) {
```

Comment:
We know this, since it is set to `true` in the first iteration of the for loop, we know that the `strandFrames` are not empty (hence at least one iteration will be done). If `standFrames` were empty, we would exit early. 
See also [here](https://github.com/ballerina-platform/ballerina-lang/pull/43035/files/64582b1321626b77767b6d0f3becb29cc94da51c#diff-c8ef2ba86a7b886642467825435303702a6179153c924c03ed3d685af82ae4f3R476-R481)

Reply:
I've removed the comment below. It only talks about the case if the `strandFrames` are empty, and this is not possible at that line.